### PR TITLE
Dat 3193 add zero state to insert infobox dialog in VE 

### PR DESCRIPTION
**Pull request by RafLeszczynski**
https://wikia-inc.atlassian.net/browse/DAT-3193

### CODE REVIEW

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 1]
```diff
@@ -155,29 +139,120 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+};
+
+/**
+ * @desc returns empty stage message
+ * @param {Array} unconvertedInfoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createEmptyState = function ( unconvertedInfoboxes ) {
+	var deferred = $.Deferred(),
+		noInfoboxesHTML = '<span class="insert-infobox-empty-state">' +
```
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: @inez: is there a better way to do it inside VE? For example some default zero state widget the will accept message url URL?

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.Dialog.scss [hunk 2]
```diff
@@ -37,3 +37,9 @@
 	margin: 5px auto 0 auto;
 }
 
+.insert-infobox-empty-state {
```
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.Dialog.scss]: I'm not sure if this is the best place for adding these styles. @inez do you have any suggestions?

**Comment by RafLeszczynski** [comment]
@kamilkoterba, @lukaszkonieczny, @adamkarminski could you please take a look a the Insights part?

File: extensions/VisualEditor/wikia/VisualEditor.i18n.php [hunk 3]
```diff
@@ -38,6 +38,9 @@
 	'wikia-visualeditor-dialogbutton-transclusion-tooltip' => 'Template',
 	'wikia-visualeditor-dialogbutton-infobox-tooltip' => 'Infobox',
 	'wikia-visualeditor-dialog-infobox-insert-title' => 'Select infobox to insert',
+	'wikia-visualeditor-dialog-infobox-insert-empty-state' => 'Sorry. No infoboxes found.',
+	'wikia-visualeditor-dialog-infobox-insert-empty-state-has-unconverted-infoboxes' => 'Convert some old infoboxes
```
Comment (sebastianmarzjan) [inline extensions/VisualEditor/wikia/VisualEditor.i18n.php]: Please avoid such newlines in messages - when parsing, this can cause unintended formatting to appear

File: extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php [hunk 4]
```diff
@@ -3,8 +3,9 @@
 
 class UnconvertedInfoboxesPage extends PageQueryPage {
 	const LIMIT = 1000;
+	const UNCONVERTED_INFOBOXES_TYPE = 'AllInfoboxes';
```
Comment (dianafa) [inline extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php]: Didn't you mean 'Nonportableinfoboxes' / 'unconvertedinfoboxes' here? :)

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+class ApiQueryUnconvertedInfoboxes extends ApiQueryBase {
+
+	const CACHE_TTL = 86400;
+	const MCACHE_KEY = 'unconvertedinfoboxes-list';
+
+	public function execute() {
+		$data = WikiaDataAccess::cache( wfMemcKey( self::MCACHE_KEY ), self::CACHE_TTL, function () {
```
Comment (kamilkoterba) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php:10]: You probably want to purge this cache once the set is changed.
Maybe here is a good place to hook in with purge of this set
https://github.com/Wikia/app/blob/dev/extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php#L41-41

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+class ApiQueryUnconvertedInfoboxes extends ApiQueryBase {
+
+	const CACHE_TTL = 86400;
```
Comment (kamilkoterba) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php]: I'd suggest using standard cache lengths
WikiaResponse::CACHE_STANDARD

File: extensions/wikia/Insights/Insights.setup.php [hunk 7]
```diff
@@ -62,6 +62,22 @@
...
 
 /**
+ * Api Querys
+ */
+$wgAutoloadClasses[ 'ApiQueryUnconvertedInfoboxes' ] = $dir . 'ApiQueryUnconvertedInfoboxes.class.php';
+
+/**
+ * Query Pages
+ */
+$wgAutoloadClasses[ 'UnconvertedInfoboxesPage' ] = $dir . 'specials/UnconvertedInfoboxesPage.class.php';
```
Comment (kamilkoterba) [inline extensions/wikia/Insights/Insights.setup.php]: UnconvertedInfoboxesPage class has separate setup file dependent on ($wgEnableInsightsExt && $wgEnableInsightsInfoboxes) config combination
https://github.com/Wikia/app/blob/dev/extensions/wikia/Insights/InsightsUnconvertedInfoboxes.setup.php#L22-22

File: extensions/wikia/Insights/Insights.setup.php [hunk 8]
```diff
@@ -62,6 +62,22 @@
 $wgAutoloadClasses['InsightsModuleController'] = $IP.'/skins/oasis/modules/InsightsModuleController.class.php';
 
 /**
+ * Api Querys
+ */
+$wgAutoloadClasses[ 'ApiQueryUnconvertedInfoboxes' ] = $dir . 'ApiQueryUnconvertedInfoboxes.class.php';
```
Comment (kamilkoterba) [inline extensions/wikia/Insights/Insights.setup.php]: Should be moved to extensions/wikia/Insights/InsightsUnconvertedInfoboxes.setup.php

File: extensions/wikia/Insights/Insights.setup.php [hunk 9]
```diff
@@ -62,6 +62,22 @@
...
+
+/**
+ * Query Pages
+ */
+$wgAutoloadClasses[ 'UnconvertedInfoboxesPage' ] = $dir . 'specials/UnconvertedInfoboxesPage.class.php';
+
+/**
+ * MW API
+ */
+$wgAPIListModules[ 'unconvertedinfoboxes' ] = 'ApiQueryUnconvertedInfoboxes';
```
Comment (kamilkoterba) [inline extensions/wikia/Insights/Insights.setup.php]: Should be moved to extensions/wikia/Insights/InsightsUnconvertedInfoboxes.setup.php

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 10]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+ve.ui.WikiaInfoboxInsertDialog.prototype.getUnconvertedInfoboxes = function () {
+	var deferred = $.Deferred();
+
+	ve.init.target.constructor.static.apiRequest( {
+		action: 'query',
+		list: 'unconvertedinfoboxes'
+	} )
+		.done( function ( data ) {
+			deferred.resolve(
+				data.query &&
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: Do you think you could make this more readable?

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 11]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+		noInfoboxesHTML = '<span class="insert-infobox-empty-state">' +
+			ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state' ) +
+			'</span>',
+		convertOldInfoboxesHTML = '<a href="' + window.location.origin +
+			'/wiki/Special:Insights/nonportableinfoboxes" class="insert-infobox-empty-state">' +
+			ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state-has-unconverted-infoboxes' ) +
+			'</a>';
+
+	deferred.resolve(
+		unconvertedInfoboxes.length === 0 ?
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: Do you think you could make this more readable?

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js (hunk 1 above)
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: VisualEditor is based on OOJS and OOJS-UI which are widget oriented frameworks (as opposed to template oriented). Most likely the way to go here would be to create a new widget (which internally in its constructor, and not only, uses jQuery to create DOM) and use that widget here.
Please refrain from concatenating HTML like this - unless there is a really good reason for it (in some cases could be performance but not in this case).

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.Dialog.scss (hunk 2 above)
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/styles/ve.ui.Dialog.scss]: Please take a look how similar things are achieved in similar Wikia's feature - maybe empty state in media dialog - and follow the pattern.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 12]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
-					data: infoboxes[i].title,
-					label:  infoboxes[i].title
-				})
-			);
-		}
-		this.select.addItems( items );
+ve.ui.WikiaInfoboxInsertDialog.prototype.createDialogContent = function ( data ) {
+	var deferred = $.Deferred(),
+		infoboxes = data.query ? data.query.allinfoboxes : [],
+		items = infoboxes.map( this.createInfoboxItemOptionWidget );
```
Comment (dianafa) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: maybe we can name this two vars in more descriptive way, e.g. `infoboxTemplates` and `infoboxTemplateOptions` ? Just a suggestion.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 13]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createInfoboxItemOptionWidget = function ( data ) {
+	return new OO.ui.DecoratedOptionWidget( {
+		data: data.title,
+		label:  data.title
+	} );
+};
+
+/**
+ * @desc creates select widget with indobox option widgets
```
Comment (dianafa) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: typo :dancers:

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+			$dbr = wfGetDB( DB_SLAVE );
+
+			return ( new WikiaSQL() )
+				->SELECT( 'qc_value', 'qc_namespace', 'qc_title' )
+				->FROM( 'querycache' )
+				->WHERE( 'qc_type' )->EQUAL_TO( UnconvertedInfoboxesPage::UNCONVERTED_INFOBOXES_TYPE )
+				->run( $dbr, function ( ResultWrapper $result ) {
+					$out = [ ];
+					while ( $row = $result->fetchRow() ) {
+						$out[] = [ 'pageid' => $row[ 'qc_value' ],
```
Comment (inez) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php]: This potentially would be easier to read:

```
$out[] = [
    'pageid' => $row[ 'qc_value' ],
    'title' => $row[ 'qc_title' ],
    'ns' => $row[ 'qc_namespace' ]
];
```

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+					}
+
+					return $out;
+				} );
+		} );
+
+		foreach ( $data as $id => $infobox ) {
+			$this->getResult()->addValue( [ 'query', 'unconvertedinfoboxes' ], null, $infobox );
+		}
+		$this->getResult()->setIndexedTagName_internal( [ 'query', 'unconvertedinfoboxes' ], 'i' );
```
Comment (inez) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php]: What does 'i' stand for? Perhaps should be commented inline as well.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 16]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+	);
+
+	return deferred.promise();
+};
+
+/**
+ * @desc adds html content to dialog
+ * @param {HTMLElement} $content
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.setDialogContent = function ( $content ) {
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js:243]: There is a mismatch between method name and method description (and what it actually do).

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 17]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+	return deferred.promise();
 };
 
 /**
+ * @desc returns empty stage message
+ * @param {Array} unconvertedInfoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createEmptyState = function ( unconvertedInfoboxes ) {
+	var deferred = $.Deferred(),
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: Why would you use promise and deferred in this method?

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 18]
```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+	if ( items.length > 0 ) {
+		this.select = this.createInfoboxSelectWidget( items );
+
+		this.select.connect( this, {
+			select: 'onInfoboxTemplateSelect'
+		} );
+		deferred.resolve( this.select.$element );
+	} else {
+		this.getUnconvertedInfoboxes()
+			.then(this.createEmptyState)
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: this.createEmptyState.bind( this ) instead of this.createEmptyState

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js (hunk 10 above)
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: will move to separate method

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js (hunk 17 above)
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: in order to keep a flat promise chain without unnecessary code indentation in `createDialogContent` method `else` statement:

``` JS
this.getUnconvertedInfoboxes()
            .then( this.createEmptyStateContent.bind( this ) )
            .then( deferred.resolve )
```

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js (hunk 1 above)
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: thanks, will create new UI widget

**Comment by RafLeszczynski** [comment]
@inez, @kamilkoterba, @dianafa could you please do the second round of CR?

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 19]
```diff
@@ -155,26 +139,112 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+ve.ui.WikiaInfoboxInsertDialog.prototype.validateGetUnconvertedInfoboxesResponseData = function ( data ) {
+	return data.query && data.query.unconvertedinfoboxes && data.query.unconvertedinfoboxes.length > 0;
+};
+
+/**
+ * @desc creates empty state dialog content
+ * @param {Array} unconvertedInfoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createEmptyStateContent = function ( unconvertedInfoboxes ) {
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js:231]: This is a follow up...
@RafLeszczynski If method doesn't need to return promise because of what it does (e.g. it's not asynchronous) then it shouldn't return promise.
In the previous comment you said that it returns promise because of the particular usage of it (together with another promise)  - from what I remember to .then in jQuery you can pass method which just return data and it should work just fine.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 20]
```diff
@@ -155,26 +139,112 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+
+		this.select.connect( this, {
+			select: 'onInfoboxTemplateSelect'
+		} );
+		deferred.resolve( this.select.$element );
+	} else {
+		// creates empty state content
+		this.getUnconvertedInfoboxes()
+			.then( this.createEmptyStateContent.bind( this ) )
+			.then( deferred.resolve )
```
Comment (dianafa) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: tests are failing due to semicolon missing here

File: extensions/VisualEditor/wikia/VisualEditor.i18n.php [hunk 21]
```diff
@@ -38,6 +38,8 @@
 	'wikia-visualeditor-dialogbutton-transclusion-tooltip' => 'Template',
 	'wikia-visualeditor-dialogbutton-infobox-tooltip' => 'Infobox',
 	'wikia-visualeditor-dialog-infobox-insert-title' => 'Select infobox to insert',
+	'wikia-visualeditor-dialog-infobox-insert-empty-state' => 'Sorry. No infoboxes found',
```
Comment (dianafa) [inline extensions/VisualEditor/wikia/VisualEditor.i18n.php:41]: just a thought - maybe should we mention here we are talking about the portable infoboxes only (and we don't care about the old ones) ?
AFAIR one of the users mentioned (in thread) that it was a surprise for him, that the new infobox edit tool in VE doesn't support the old ones.

Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/VisualEditor.i18n.php:41]: @dianafa: I will ask @shareif about this one.

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+					return $out;
+				} );
+		} );
+
+		foreach ( $data as $id => $infobox ) {
+			$this->getResult()->addValue( [ 'query', 'unconvertedinfoboxes' ], null, $infobox );
+		}
+
+		// tag name used in XML format API response
+		$xmlTagName = 'infobox';
```
Comment (idradm) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php]: You could const that

**Comment by kamilkoterba** [comment]
Changes to Insights ext LGTM!

File: extensions/VisualEditor/wikia/VisualEditor.i18n.php (hunk 21 above)
Comment (RafLeszczynski) [inline extensions/VisualEditor/wikia/VisualEditor.i18n.php:41]: Created separate issue in Jira: https://wikia-inc.atlassian.net/browse/DAT-3235

**Comment by idradm** [comment]
LGTM

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 23]
```diff
@@ -155,26 +139,112 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+	return deferred.promise();
+};
+
+/**
+ * @desc gets list of unconverted infoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getUnconvertedInfoboxes = function () {
+	var deferred = $.Deferred(),
+		self = this;
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: In VisualEditor we use .bind instead of self trick, e.g.:

```
function ( data ) {
}.bind( this )
```

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 24]
```diff
@@ -155,26 +139,112 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+		this.select.connect( this, {
+			select: 'onInfoboxTemplateSelect'
+		} );
+
+		deferred.resolve( this.select.$element );
+	} else {
+		// creates empty state content
+		this.getUnconvertedInfoboxes().then(
+			( function ( data ) {
+				deferred.resolve( this.createEmptyStateContent( data ).bind( this ) );
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: I don't think bind here is correctly used - you call bind on the return call to createEmptyStateContent method.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 25]
```diff
@@ -155,26 +139,112 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
...
+			} ).bind( this )
+		);
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: I find those few lines above little bit hard to follow, if you want to do one thing (getUnconvertedInfoboxes) then do other thing (createEmptyStateContent) and then do another thing (resolve) why not just use then two times? Remember that the output data from first then will be carried over to second then. I actually created this simple example for you so you can see how it works exactly: https://jsfiddle.net/Inez/ggu74j6c/

Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: Actually something like this ~should work:

```
this.getUnconvertedInfoboxes()
    .then(this.createEmptyStateContent.bind(this))
    .then(deferred.resolve);
```

**Comment by RafLeszczynski** [comment]
@inez: sorry for misunderstanding. I thought you didn't want to use `.them()` chaining for synchronous functions. Hope this will be the last round of CR. Thanks for all comments

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js [hunk 26]
```diff
@@ -131,7 +115,7 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.getInfoboxTemplates = function () {
 			} );
 		this.gettingTemplateNames = deferred.promise();
 	}
-	return this.gettingTemplateNames;
+	return deferred.promise();
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js]: On the second call to getInfoboxTemplates (so when this.gettingTemplateNames is set) this code will attempt to call method promise() on undefined (the deferred).
Btw. why to make this (return different thing) in the first place?

File: extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaInsertInfoboxEmptyStateWidget.js [hunk 27]
```diff
@@ -0,0 +1,36 @@
...
+ * @extends OO.ui.Widget
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInsertInfoboxEmptyState = function VeUiWikiaInsertInfoboxEmptyState( config ) {
+	var noInfoboxFoundMsg = ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state'),
+		convertInfoboxesMsg = ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state-has-unconverted-infoboxes'),
+
+		insigthsURL = window.location.origin + '/wiki/Special:Insights/nonportableinfoboxes',
```
Comment (inez) [inline extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaInsertInfoboxEmptyStateWidget.js]: I recommend you use mw.Title (and mw.Uri) to build such URLs.

File: extensions/wikia/Insights/InsightsUnconvertedInfoboxesHooks.class.php [hunk 28]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+class InsightsUnconvertedInfoboxesHooks {
+	static public function onUnconvertedInfoboxesQueryRecached() {
+		F::app()->wg->Memc->delete( wfMemcKey( ApiQueryUnconvertedInfoboxes::MCACHE_KEY ) );
```
Comment (inez) [inline extensions/wikia/Insights/InsightsUnconvertedInfoboxesHooks.class.php:5]: I thought we are not using the F anymore and just $wgMemc->delete.. @owend What do you say?

File: extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php [hunk 29]
```diff
@@ -80,6 +81,8 @@ public function recache( $limit = false, $ignoreErrors = true ) {
 			}
 		}
 
+		wfRunHooks( 'UnconvertedInfoboxesQueryRecached' );
```
Comment (inez) [inline extensions/wikia/Insights/specials/UnconvertedInfoboxesPage.class.php:84]: For me hard to say what's the purpose of this code and functionality in context of Dat 3193.

File: extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php [hunk 30]
```diff
@@ -0,0 +1,41 @@
...
+							'title' => $row[ 'qc_title' ],
+							'ns' => $row[ 'qc_namespace' ]
+						];
+					}
+
+					return $out;
+				} );
+		} );
+
+		foreach ( $data as $id => $infobox ) {
```
Comment (inez) [inline extensions/wikia/Insights/ApiQueryUnconvertedInfoboxes.class.php]: No need to declare $id as it is not being used anywhere.

**Comment by RafLeszczynski** [comment]
Passing to QA. Any other potential code optimisations will be addressed in separate Jira issues and PRs
